data-model/src/main/scala/za/co/absa/enceladus/model/dataFrameFilter/package.scala

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.model
+
+import com.fasterxml.jackson.annotation.JsonSubTypes.Type
+import com.fasterxml.jackson.annotation.{JsonIgnore, JsonSubTypes, JsonTypeInfo}
+import org.apache.spark.sql.Column
+import org.apache.spark.sql.functions._
+
+package object dataFrameFilter {
```

Comment:
It's a comparison to a constant. But you are right, a `IsNullFilter` sounds useful.

Reply:
I know it is a constant but `===` is not NullSafe. You have no way of allowing/havin Null now. You would need to use `<=>` which is NullSafe

Reply:
You are right. And originally the `value` was `String` so it didn't make sense. But now it's `Any` so it can be `null`.
So yeah, I would change to `<=>`. 
Two questions raise then:
* Name of the `EqualsFilter`
* create an `IsNullFilter` for verbosity or not needed

Reply:
You might have noticed I went a little bit different way. Adding type (to avoid the need of Spark quessing the type, and all kind of type information loss between JSON and Scala), enforcing the value to be string and adding the `IsNullFilter`.
That should still cover all the possible filtering options.